### PR TITLE
smp: fix multiple pipelines on different cores

### PR DESCRIPTION
**Pull request by tlauda**
Series of fixes for streaming pipelines on different cores.

### CODE REVIEW

File: src/platform/icelake/include/platform/shim.h [hunk 1]
```diff
@@ -161,7 +161,7 @@
 
 #define SHIM_PWRCTL		0x90
 #define SHIM_PWRSTS		0x92
-#define SHIM_PWRCTL_TCPDSP0PG	(0x1 << 0)
+#define SHIM_PWRCTL_TCPDSPPG(x)	BIT(x)
```
Comment (jajanusz) [inline src/platform/icelake/include/platform/shim.h]: need it also for suecreek

**Comment by tlauda** [comment]
@xiulipan Can you take a look why the device tests are failing on topology load?

**Comment by xiulipan** [comment]
@tlauda 
And UP2 Nocodec is some test framework issue. Re trigger for an other test.

For BYT there are IPC timeouts
```
[   16.756239 <    0.000013>] sof-audio-acpi 80860F28:01: ipc tx: 0x30010000: GLB_TPLG_MSG: COMP_NEW
[   17.059541 <    0.303302>] sof-audio-acpi 80860F28:01: error: ipc timed out for 0x30010000 size 76
[   17.059691 <    0.000150>] sof-audio-acpi 80860F28:01: error: unexpected fault 0x70028800 trace 0x30010000
[   17.059783 <    0.000092>] sof-audio-acpi 80860F28:01: error: DSP failed to add widget id 0 type 23 name : PCM0P stream Low Latency Playback 0 reply 0
[   17.062625 <    0.002842>] sof-audio-acpi 80860F28:01: ASoC: failed to load widget PCM0P
```

**Comment by tlauda** [comment]
@xiulipan Do we have any FW logs? I have no idea why it's failing on BYT during topology load. There are no potential changes that could cause this.

**Comment by xiulipan** [comment]
@tlauda 
The error log I can got is empty. Maybe need some workaround to debug.

And the DMA trace init is failed.

https://sof-ci.01.org/sofpr/PR1348/build1863/devicetest/BYT_MB_RT5651/verify_pcm_list.sh/dmesg.txt
```
[    8.035755 <    0.000005>] sof-audio-acpi 80860F28:01: ipc tx: 0x90010000: GLB_TRACE_MSG
[    8.338175 <    0.302420>] sof-audio-acpi 80860F28:01: error: ipc timed out for 0x90010000 size 40
[    8.338454 <    0.000279>] sof-audio-acpi 80860F28:01: error: unexpected fault 0x70028800 trace 0x90010000
[    8.338696 <    0.000242>] sof-audio-acpi 80860F28:01: error: can't set params for DMA for trace -110
[    8.338987 <    0.000291>] sof-audio-acpi 80860F28:01: warning: failed to initialize trace -110
```

**Comment by tlauda** [comment]
@lgirdwood BYT issue fixed, ready to be merged.

**Comment by lyakh** [comment]
@tlauda @plbossart I have a question to the last commit in this series: "dw-dma: handle interrupts per core" - do I understand correctly, that with this change different DMA channels on the same DW DMA controllers can be used on different cores even on platforms, where CONFIG_DMA_AGGREGATED_IRQ is set? But that would mean that different cores access and modify common `struct dma_pdata` objects:
```
	p->mask_irq_channels[cpu] |= BIT(channel);
```
even though those masks are per-CPU, they are just 32 bits long, so 8 of them will share a common cache line. So when the cache is written back, it will overwrite those masks for neighbour channels. In fact the `chan[]` array isn't cache-line aligned either...

**Comment by tlauda** [comment]
@lyakh See line 1216. The whole struct is allocated in uncached region to avoid multiple invalidations and writebacks across the whole driver. Also every allocation done by alloc api is aligned to cache line by default (not explicitly), so we can omit any additional alignments.

**Comment by lyakh** [comment]
@tlauda oh, indeed, I missed the UNCACHED flag, thanks. Do we have an idea how much slower access to uncacheable RAM is? Maybe we should use the same for interrupt controller descriptors...

**Comment by tlauda** [comment]
@lyakh We don't have exact numbers. We would need to compare in the specific scenario uncached vs writeback and invalidate. There is a huge possibility, that we are not wasting many cycles.

**Pull request merged**